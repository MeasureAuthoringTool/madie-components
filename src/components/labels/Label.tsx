import React from "react";
import tw, { styled } from "twin.macro";

export interface LabelProps
  extends React.LabelHTMLAttributes<HTMLLabelElement> {
  text: string;
  isOptional?: boolean;
}

const StyledLabel = styled.label(({}) => [
  tw`block text-sm font-medium text-gray-700`,
]);

const StyledSpan = styled.span(({}) => [tw`text-sm text-gray-500`]);

export function LabelComponent(props: LabelProps) {
  const { isOptional, ...args } = props;
  return (
    <div tw="flex justify-between">
      <StyledLabel {...args}>{props.text}</StyledLabel>
      {!!isOptional && <StyledSpan>Optional</StyledSpan>}
    </div>
  );
}
