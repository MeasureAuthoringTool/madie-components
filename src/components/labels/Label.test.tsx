import * as React from "react";
import { render } from "@testing-library/react";
import Label from "./Label";

const testId = "label-component-test";
const labelText = "Label";
const cornerHint = "Optional";

describe("Label Component", () => {
  it("should render a Label with no corner hint", () => {
    const { getByTestId } = render(<Label text={labelText} />);
    const label = getByTestId(testId);
    expect(label).toBeInTheDocument();
    expect(label).toHaveTextContent(labelText);
    expect(label).toMatchSnapshot();
  });

  it("should render a Label with corner hint", () => {
    const { getByTestId } = render(
      <Label text={labelText} cornerHint={cornerHint} />
    );
    const label = getByTestId(testId);
    expect(label).toBeInTheDocument();
    expect(label).toHaveTextContent(labelText);
    expect(label).toHaveTextContent(cornerHint);
    expect(label).toMatchSnapshot();
  });
});
