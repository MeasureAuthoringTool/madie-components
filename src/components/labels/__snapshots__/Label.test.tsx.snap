// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render a Label with corner hint 1`] = `
<div
  class="Label___StyledDiv-sc-goa2uk-0 bFBXIw"
  data-testid="label-component-test"
>
  <label
    class="Label___StyledLabel-sc-goa2uk-1 lFRji"
  >
    Label
  </label>
  <span
    class="Label___StyledSpan-sc-goa2uk-2 ftEbCx"
  >
    Optional
  </span>
</div>
`;

exports[`Label Component should render a Label with no corner hint 1`] = `
<div
  class="Label___StyledDiv-sc-goa2uk-0 bFBXIw"
  data-testid="label-component-test"
>
  <label
    class="Label___StyledLabel-sc-goa2uk-1 lFRji"
  >
    Label
  </label>
</div>
`;
